File: src/Models/Show.cs
namespace CineComplex.Models
{
    public class Show
    {
        private int _showId;//Must be done internally from database
        private int _movieId;
        private int _theatreId;
        private decimal _platinumSeatRate;
        private decimal _goldSeatRate;
        private decimal _silverSeatRate;
        private DateTime _startDate;
        private DateTime _endDate;
        private static Dictionary<int, Show>? _shows;

        #region Dictionaries

        public static Dictionary<int, Show>? Shows { get => _shows; set => _shows = value; }


        #endregion

        public int ShowId { get => _showId; set => _showId = value; }
        public int MovieId { get => _movieId; set => _movieId = value; }
        public int TheatreId { get => _theatreId; set => _theatreId = value; }
        public decimal PlatinumSeatRate { get => _platinumSeatRate; set => _platinumSeatRate = value; }
        public decimal GoldSeatRate { get => _goldSeatRate; set => _goldSeatRate = value; }
        public decimal SilverSeatRate { get => _silverSeatRate; set => _silverSeatRate = value; }
        public DateTime StartDate { get => _startDate; set => _startDate = value; }
        public DateTime EndDate { get => _endDate; set => _endDate = value; }

        public Show(int movieId, int theatreId, DateTime startDate, DateTime endDate, decimal platinumRate, decimal goldRate, decimal silverRate)
        {
            MovieId = movieId;
            TheatreId = theatreId;
            StartDate = startDate;
            EndDate = endDate;
            PlatinumSeatRate = platinumRate;
            GoldSeatRate = goldRate;
            SilverSeatRate = silverRate;
            ShowId = 10101;// Constructor must generate from database
        }

        public void DisplayShowDetails()
        {
            Console.WriteLine($"Show ID : {ShowId}");
            Console.WriteLine($"Movie ID : {MovieId}");
            Console.WriteLine($"Theatre ID : {TheatreId}");
            Console.WriteLine($"Platinum Seat Rate : {PlatinumSeatRate}");
            Console.WriteLine($"Gold Seat Rate : {GoldSeatRate}");
            Console.WriteLine($"Silver Seat Rate : {SilverSeatRate}");
        }
    }
}

File: src/Models/Show.py
# from sqlalchemy import Column, Integer, DateTime, ForeignKey
#
# from src.Classes.Base.ABaseModel import ABaseModel
#
#
# class Show(ABaseModel):
#     __tablename__ = "shows"
#
#     id = Column(Integer, primary_key=True, index=True)
#     movie_id = Column(Integer, ForeignKey("movies.id"), nullable=False)
#     theatre_id = Column(Integer, ForeignKey("theatres.id"), nullable=False)
#     start_date = Column(DateTime, nullable=False)
#     end_date = Column(DateTime, nullable=False)
#     platinum_seat_rate = Column(Integer, nullable=False)
#     gold_seat_rate = Column(Integer, nullable=False)
#     silver_seat_rate = Column(Integer, nullable=False)
#
#     def __repr__(self):
#         return f"Show(ID={self.id}, MovieID={self.movie_id}, TheatreID={self.theatre_id})"
